miranda/convert/_data_corrections.py

```diff
@@ -171,19 +171,29 @@ def threshold_mask(
     else:
         mask_variable = mask.name
 
-    log_msg = f"Masking dataset with {mask_variable}."
-    if mask_cutoff:
-        log_msg = f"{log_msg.strip('.')} at `{mask_cutoff}` cutoff value."
-    logging.info(log_msg)
+    try:
+        from clisops.core import subset_bbox
+
+        log_msg = f"Masking dataset with {mask_variable}."
+        if mask_cutoff:
+            log_msg = f"{log_msg.strip('.')} at `{mask_cutoff}` cutoff value."
+        logging.info(log_msg)
 
-    lon_bounds = np.array([ds.lon.min(), ds.lon.max()])
-    lat_bounds = np.array([ds.lat.min(), ds.lat.max()])
+        lon_bounds = np.array([ds.lon.min(), ds.lon.max()])
+        lat_bounds = np.array([ds.lat.min(), ds.lat.max()])
 
-    mask_subset = subset_bbox(
-        mask,
-        lon_bnds=lon_bounds,
-        lat_bnds=lat_bounds,
-    ).load()
+        mask_subset = subset_bbox(
+            mask,
+            lon_bnds=lon_bounds,
+            lat_bnds=lat_bounds,
+        ).load()
+    except ModuleNotFoundError:
+        log_msg = (
+            "This function requires the `clisops` library which is not installed. "
+            "subsetting step will be skipped."
+        )
+        warnings.warn(log_msg)
+        mask_subset = mask.load()
```

Comment:
Below, we do `ds.where(mask_subset.notnull())` . This assumes `ds` and `mask` have the same coordinates.

I tested it and `ds.where(mask)` will only contain the area where both overlap. Thus, I'm not sure `subset_bbox` is needed ?  Unless, this was needed to reduce the RAM needed by loading the mask. The whole try/except seems to be skippable.